File: go/elons-toys/elons_toys.go
package elon

import "fmt"

func (c *Car) Drive() {
	if c.battery >= c.batteryDrain {
		c.battery -= c.batteryDrain
		c.distance += c.speed
	}
}

// TODO: define the 'DisplayDistance() string' method
func (c Car) DisplayDistance() string {
	//panic("")
	return "Driven " + fmt.Sprintf("%d", c.distance) + " meters"
}

// TODO: define the 'DisplayBattery() string' method
func (c Car) DisplayBattery() string {
	//panic("")
	return "Battery at " + fmt.Sprintf("%d", c.battery) + "%"
}

// TODO: define the 'CanFinish(trackDistance int) bool' method
func (c Car) CanFinish(trackDistance int) bool {
	//panic("")
	return (c.battery / c.batteryDrain * c.speed) >= trackDistance
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
//
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more
// functionality.

File: csharp/elons-toys/ElonsToys.cs
using System;

class RemoteControlCar{

    public int meters = 0;
    public double battery = 100.0;

    public static RemoteControlCar Buy() => new RemoteControlCar();

    public string DistanceDisplay() =>  "Driven "+meters+" meters";

    public string BatteryDisplay() => (battery>0) ? "Battery at "+battery+"%" : "Battery empty";

    public void Drive()
    {
        if (battery == 0)
        {
            return;
        }

        meters += 20;
        battery -= 1;
    }
}
